### PR TITLE
Rename `MLflow Deployments Server` to `MLflow AI Gateway` in docs

### PR DESCRIPTION
**Pull request by harupy**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/harupy/mlflow/pull/13124?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/13124/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 13124
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

title

### How is this PR tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

`Yes` should be selected for bug fixes, documentation updates, and other small changes. `No` should be selected for new features and larger changes. If you're unsure about the release classification of this PR, leave this unchecked to let the maintainers decide.

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Bug fixes, doc updates and new features usually go into minor releases.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Bug fixes and doc updates usually go into patch releases.

</details>

- [ ] Yes (this PR will be cherry-picked and included in the next patch release)
- [x] No (this PR will be included in the next minor release)

### CODE REVIEW

File: docs/source/llms/prompt-engineering/index.rst [hunk 1]
```diff
@@ -20,27 +20,27 @@ Quickstart
 
 The following guide will get you started with MLflow's UI for prompt engineering.
 
-Step 1: Create an MLflow Deployments Server Completions or Chat Endpoint
+Step 1: Create an MLflow AI Gateway Completions or Chat Endpoint
 ------------------------------------------------------------------------
-To use the prompt engineering UI, you need to create one or more :ref:`MLflow Deployments Server <deployments>`
+To use the prompt engineering UI, you need to create one or more :ref:`MLflow AI Gateway <deployments>`
```
Comment (harupy) [inline docs/source/llms/prompt-engineering/index.rst:25]:
```suggestion
To use the prompt engineering UI, you need to create one or more :ref:`MLflow AI Gateway <gateway>`
```

File: docs/source/llms/deployments/index.rst
Comment (harupy) [inline docs/source/llms/deployments/index.rst:1]: rename this file to `docs/source/llms/gateway/index.rst`.

File: docs/source/llms/deployments/index.rst [hunk 2]
```diff
@@ -170,7 +170,7 @@ Step 8: Compare Provider Models
 -------------------------------
 Here's an example of adding a new model from a provider to determine which model instance is better for a given use case.
 
-Firstly, update the :ref:`MLflow Deployments Server config <deployments_configuration>` YAML file with the additional endpoint definition to test:
+Firstly, update the :ref:`MLflow AI Gateway config <deployments_configuration>` YAML file with the additional endpoint definition to test:
```
Comment (BenWilson2) [inline docs/source/llms/deployments/index.rst:173]: Double check that we're not renaming the alias link `deployments_configuration`

File: docs/source/llms/deployments/index.rst [hunk 3]
```diff
@@ -424,15 +424,15 @@ Conversely, for generating embeddings of text, you would choose an embedding mod
...
-The MLflow Deployments Server relies on a user-provided configuration file, written in YAML,
+The MLflow AI Gateway relies on a user-provided configuration file, written in YAML,
 that defines the endpoints and providers available to the server. The configuration file dictates
-how the deployments server interacts with various language model providers and determines the end-points that
+how the gateway server interacts with various language model providers and determines the end-points that
 users can access.
 
-Deployments Server Configuration
+gateway server Configuration
 --------------------------------
```
Comment (BenWilson2) [inline docs/source/llms/deployments/index.rst:436]:
```suggestion
AI Gateway server Configuration
-------------------------------
```

File: docs/source/llms/deployments/index.rst [hunk 4]
```diff
@@ -488,10 +488,10 @@ environment. For example:
 
 .. _deployments_configuration_details:
 
-Deployments Server Configuration Details
+gateway server Configuration Details
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (BenWilson2) [inline docs/source/llms/deployments/index.rst:492]:
```suggestion
AI Gateway server Configuration Details
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: docs/source/llms/deployments/index.rst [hunk 5]
```diff
@@ -756,19 +756,19 @@ For specifying an API key, there are three options:
 
 .. _deployments_query:
 
-Querying the Deployments Server
+Querying the gateway server
 ===============================
```
Comment (BenWilson2) [inline docs/source/llms/deployments/index.rst:760]:
```suggestion
Querying the AI Gateway server
==============================
```

File: docs/source/llms/prompt-engineering/index.rst [hunk 6]
```diff
@@ -20,27 +20,27 @@ Quickstart
...
 completions or chat :ref:`Endpoints <deployments-endpoints>`. Follow the
-:ref:`MLflow Deployments Server Quickstart guide <deployments-quickstart>` to easily create an endpoint in less than five
-minutes. If you already have access to an MLflow Deployments Server endpoint of type ``llm/v1/completions``
+:ref:`MLflow AI Gateway Quickstart guide <deployments-quickstart>` to easily create an endpoint in less than five
```
Comment (BenWilson2) [inline docs/source/llms/prompt-engineering/index.rst:27]: Are we renaming the alias hyperlink?

Comment (harupy) [reply docs/source/llms/prompt-engineering/index.rst:27]: Do we need to?

**Review by BenWilson2 — APPROVED** [review]
Overall, looks good. There are some overhangs on the header section symbols and some validations for links, but other than those nits, LGTM!

File: docs/source/llms/prompt-engineering/index.rst (hunk 6 above)
Comment (BenWilson2) [reply docs/source/llms/prompt-engineering/index.rst:27]: we don't need to. That was just a confirmation to make sure it was intentional to leave it :)

File: docs/source/llms/prompt-engineering/index.rst (hunk 1 above)
Comment (harupy) [reply docs/source/llms/prompt-engineering/index.rst:25]: We won't do this.

File: docs/source/llms/deployments/index.rst
Comment (BenWilson2) [reply docs/source/llms/deployments/index.rst:1]: Can we register a redirect if we're renaming a page? https://github.com/mlflow/mlflow/blob/afe91359853aed8fe32a57df603dea7f6cfc0f9f/docs/source/conf.py#L51-L53

**Comment by BenWilson2** [comment]
Reminder for this: https://github.com/mlflow/mlflow/pull/13124#discussion_r1758062293

File: docs/source/llms/deployments/index.rst
Comment (harupy) [reply docs/source/llms/deployments/index.rst:1]: I won't rename pages.

**Pull request merged**